chia/rpc/wallet_rpc_api.py

```diff
@@ -3457,161 +3427,116 @@ async def nft_get_info(self, request: dict[str, Any]) -> EndpointResult:
-            if not isinstance(meta["uris"], list):
-                return {"success": False, "error": "Data URIs must be a list"}
-            if not isinstance(meta.get("meta_uris", []), list):
-                return {"success": False, "error": "Metadata URIs must be a list"}
-            if not isinstance(meta.get("license_uris", []), list):
-                return {"success": False, "error": "License URIs must be a list"}
+        for meta in request.metadata_list:
             nft_metadata = [
-                ("u", meta["uris"]),
-                ("h", hexstr_to_bytes(meta["hash"])),
-                ("mu", meta.get("meta_uris", [])),
-                ("lu", meta.get("license_uris", [])),
-                ("sn", uint64(meta.get("edition_number", 1))),
-                ("st", uint64(meta.get("edition_total", 1))),
+                ("u", meta.uris),
+                ("h", meta.hash),
+                ("mu", meta.meta_uris),
+                ("lu", meta.license_uris),
+                ("sn", meta.edition_number),
+                ("st", meta.edition_total),
             ]
-            if "meta_hash" in meta and len(meta["meta_hash"]) > 0:
-                nft_metadata.append(("mh", hexstr_to_bytes(meta["meta_hash"])))
-            if "license_hash" in meta and len(meta["license_hash"]) > 0:
-                nft_metadata.append(("lh", hexstr_to_bytes(meta["license_hash"])))
+            if meta.meta_hash is not None:
+                nft_metadata.append(("mh", meta.meta_hash))
+            if meta.license_hash is not None:
+                nft_metadata.append(("lh", meta.license_hash))
             metadata_program = Program.to(nft_metadata)
             metadata_dict = {
                 "program": metadata_program,
-                "royalty_pc": royalty_percentage,
+                "royalty_pc": request.royalty_percentage,
                 "royalty_ph": royalty_puzhash,
             }
             metadata_list.append(metadata_dict)
-        target_address_list = request.get("target_list", None)
-        target_list = []
-        if target_address_list:
-            for target in target_address_list:
-                target_list.append(decode_puzzle_hash(target))
-        mint_number_start = request.get("mint_number_start", 1)
-        mint_total = request.get("mint_total", None)
-        xch_coin_list = request.get("xch_coins", None)
-        xch_coins = None
-        if xch_coin_list:
-            xch_coins = {Coin.from_json_dict(xch_coin) for xch_coin in xch_coin_list}
-        xch_change_target = request.get("xch_change_target", None)
-        if xch_change_target is not None:
-            if xch_change_target[:2] == "xch":
-                xch_change_ph = decode_puzzle_hash(xch_change_target)
+        target_list = [decode_puzzle_hash(target) for target in request.target_list]
+        if request.xch_change_target is not None:
+            if request.xch_change_target[:2] == "xch":
```

Comment:
i think we have a (pre-existing) bug here.  this can never be true.  (also why having a parser and checking the hrp or whatever is better than string processing `:]`)

```python-console
Python 3.12.7 (main, Nov 20 2024, 20:26:30) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> "xch"[:2]
'xc'
```

Reply:
ticket made: https://chia-network.atlassian.net/browse/CHIA-3206